### ISSUE TITLE
`cast` should parse/format returned values instead of showing everything as hex

### ISSUE THREAD

**Issue by mds1** [foundry-rs/foundry#1193]
### Component

Cast

### Describe the feature you would like

Notice how seth displays decimal for things like block numbers, whereas cast shows everything in hex with unnecessary quotes around it. I'm not sure if the formatting is part of the JSON RPC spec, or custom, but this seems to apply to most `cast commands`

### `cast block`

```
$ cast block latest timestamp
0x624b1157
```

```
$ seth block latest timestamp
1649086807
```

### `cast tx`

```
$ cast tx 0xbedc92801b8a7128898f484006f9e9fb795630f1826f35dd3544db38c07594ab --rpc-url $ARBITRUM_RPC_URL
blockHash            "0x918b0fbe7235f5a058890653f82c5e483ff0bbd1c84668e930e19b424e034540"
blockNumber          "0x8a4058"
from                 "0x22222b7c879e6c09690d1e5eb10437e68987882c"
gas                  "0x0"
gasPrice             "0x0"
hash                 "0xbedc92801b8a7128898f484006f9e9fb795630f1826f35dd3544db38c07594ab"
input                "0x679b6ded00000000000000000000000022222b7c879e6c09690d1e5eb10437e68987882c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000411479445100000000000000000000000022222b7c879e6c09690d1e5eb10437e68987882c00000000000000000000000022222b7c879e6c09690d1e5eb10437e68987882c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000"
nonce                "0x0"
r                    "0x0"
s                    "0x0"
to                   "0x000000000000000000000000000000000000006e"
transactionIndex     "0x5"
v                    "0x0"
value                "0xb1a2bc2ec50000"
```

```
$ seth tx 0xbedc92801b8a7128898f484006f9e9fb795630f1826f35dd3544db38c07594ab --rpc-url $ARBITRUM_RPC_URL
arbSubType              null
arbType                 9
blockHash               0x918b0fbe7235f5a058890653f82c5e483ff0bbd1c84668e930e19b424e034540
blockNumber             9060440
from                    0x22222b7c879e6c09690d1e5eb10437e68987882c
gas                     0
gasPrice                0
hash                    0xbedc92801b8a7128898f484006f9e9fb795630f1826f35dd3544db38c07594ab
indexInParent           null
input                   0x679b6ded00000000000000000000000022222b7c879e6c09690d1e5eb10437e68987882c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000411479445100000000000000000000000022222b7c879e6c09690d1e5eb10437e68987882c00000000000000000000000022222b7c879e6c09690d1e5eb10437e68987882c0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000000
l1BlockNumber           14502213
l1SequenceNumber        0x80000000000000000000000000000000000000000000000000000000000602b5
nonce                   0
parentRequestId         null
r                       0
s                       0
to                      0x000000000000000000000000000000000000006e
transactionIndex        5
v                       0
value                   50000000000000000
```

**Comment by Bind** [comment]
Hallo, starting to poke around the foundry code base a bit more and this seems like a good issue to tackle. 

[$ cast tx ...](https://github.com/gakonst/foundry/blob/6130af7d3fd581c22b42d232011fa1576bfc6ac0/cast/src/lib.rs#L520) 
Seems the problem is the `Transaction` type is casted to a json `Value` which erases the typing information and then the json gets printed.  That typing information should drive the logic for rendering. Simplest solution would be to use a trait. 

My approach would be something like...
- [ ] trait toTable{...} (or some other name)
- [ ] impl toTable for Transaction{...}
- [ ] replace [else statement](https://github.com/gakonst/foundry/blob/6130af7d3fd581c22b42d232011fa1576bfc6ac0/cast/src/lib.rs#L542) with `transaction_result.toTable()`
- [ ] Reorganize json parsing to still respect `to_json` flag

Let me know if this seems reasonable and I'll have something submitted soon!

**Comment by onbjerg** [comment]
Was this fixed in #1253? (thanks @Bind!)

**Comment by mds1** [comment]
Yep, closing! 

Thanks for implementing @Bind 🙌
